dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPITest.java

```diff
@@ -1047,54 +1068,170 @@ public void testSaveFieldVariable_KeyWithSpecialChars_ShouldSucceed(final String
+        caseModUser.shouldCreateNewVariable = true;
+
+        return new GraphQLFieldNameCompatibilityTestCase[] {
+                caseModDateIncompatible,
+                caseModDateCompatible,
+                caseTitleCompatible,
+                caseTitleIncompatible,
+                caseTitleImage,
+                caseContentType,
+                caseBaseType,
+                caseLive,
+                caseWorking,
+                caseArchived,
+                caseLocked,
+                caseConLanguage,
+                caseIdentifier,
+                caseInode,
+                caseHost,
+                caseFolder,
+                caseUrlMap,
+                caseOwner,
+                caseModUser
+        };
     }
 
     @Test
     @UseDataProvider("dataProviderGraphQLReservedNames")
-    public void test_SaveFieldWithReservedGraphqlName_ShouldSuffixConsecutiveToVariable(
-            final String fieldName)
+    public void test_SaveFieldWithIncompatibleGraphQLNameShouldUseDifferentName(
+            final GraphQLFieldNameCompatibilityTestCase testCase)
             throws DotSecurityException, DotDataException {
 
         final ContentType type = new ContentTypeDataGen().nextPersisted();
         try {
-            Field field1 = FieldBuilder.builder(TextField.class)
-                    .name(fieldName)
-                    .contentTypeId(type.id())
-                    .indexed(false)
-                    .listed(false)
-                    .fixed(true)
-                    .build();
-            field1 = fieldAPI.save(field1, user);
-
-            Assert.assertNotNull(field1);
-            Assert.assertTrue(UtilMethods.isSet(field1.variable()));
-            Assert.assertNotEquals(fieldName, field1.variable());
-
-            // let's create a new field to make sure it's getting a new variable
-
-            Field field2 = FieldBuilder.builder(TextField.class)
-                    .name(fieldName)
-                    .contentTypeId(type.id())
-                    .indexed(false)
-                    .listed(false)
-                    .fixed(true)
-                    .build();
-            field2 = fieldAPI.save(field2, user);
-
-            Assert.assertNotNull(field2);
-            Assert.assertTrue(UtilMethods.isSet(field2.variable()));
-            Assert.assertNotEquals(fieldName, field2.variable());
-
-            // let's compare the two field vars are different
-
-            Assert.assertNotEquals(field1.variable(), field2.variable());
-
-
+            final ContentType contentType = new ContentTypeDataGen().nextPersisted();
+            final Field field = createField(contentType, testCase.fieldName,
+                    testCase.fieldType);
+
+            if(testCase.shouldCreateNewVariable) {
+                Assert.assertNotEquals(testCase.fieldName, field.variable());
+            } else  {
+                Assert.assertEquals(testCase.fieldName, field.variable());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4995822)